lib/auth0/appEndpointHandler.js

```diff
@@ -6,11 +6,15 @@ module.exports = function (webtask, options, ctx, req, res, routingInfo) {
     function authenticate() {
         var token = options.getAccessToken(ctx, req);
         if (!token) {
+            var stateInfo = {
+                appPath: routingInfo.appPath,
+                query: routingInfo.query
+            };
             return options.loginError({
                 code: 401,
                 message: 'Unauthorized.',
                 error: 'Missing access token.',
-                redirect: routingInfo.baseUrl + '/login'
+                redirect: routingInfo.baseUrl + '/login?state=' + encodeURIComponent(JSON.stringify(stateInfo))
```

Comment:
In OAuth2, it is my understanding that the `state` parameter should be (or contain) an unguessable identifier that uniquely identifies the request to the client's (browser) authentication state. This is used to prevent CSRF attacks.

It also seems that because the ultimate redirect is encoded as an unsigned value in the url, a webtask using this authn/authz tooling would be vulnerable to open redirection issues.

One experimental approach I came across to mitigate these issues is described at https://tools.ietf.org/html/draft-bradley-oauth-jwt-encoded-state-00 whereby the `state` parameter will contain a signed jwt w/ the same information as `stateInfo`.

What do you think?